### PR TITLE
Enable table column resize feature in GHS

### LINKED ISSUE #12191: Table column resize doesn't work well with GHS
## 📝 Provide detailed reproduction steps (if any)

1.  Add `TableColumnResize` to http://localhost:8125/ckeditor5-html-support/tests/manual/ghs-all-features.html.
2.  Open the manual test.
3.  Insert a table and resize the columns.
4.  Run `editor.setData( editor.getData() )`.

### ✔️ Expected result

The column remains resized.

### ❌ Actual result

The column is not resized.

### Notes

It works fine after removing GHS.

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by Dumluregn**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Fix (table,html-support): Fixed table column resize integration with General HTML Support. Closes #12191.

---

### Additional information

It's a two-parts fix:

1.  On the GHS side, it adds the `colgroup` and `col` elements schema. It resolves the issue where these elements were filtered out as `unsafe`. Please note that it only happened after using `editor.setData()` - if you just open a demo and insert a table, it can be resized normally.
2.  On the table column resize feature the `ck-table-resized` CSS class handling had to be improved as it was not always present when it should be. It was causing incorrect table rendering during the upcast in some cases (e.g. for a resized table with a lot of text in cell). Also it didn't work well for a table with `<colgroup>` element but without `width` attribute in the editor that was turned into read-only mode during resizing - the class was removed while it should stay intact (there is a unit test for that case - `if tableWidth was set for the first time`).

### Testing

PR should be tested against the epic branch it is targeted into - [ck/**epic**/11857-table-resize-improvements](https://github.com/ckeditor/ckeditor5/compare/ck/epic/11857-table-resize-improvements...ck/12191-table-resize-ghs?expand=1). **Please note that the behaviour on that branch can be different than on the `master` one, but it is still broken.** 

I created a separate branch to facilitate testing: `ck/12191-table-resize-ghs-testing`. Changes:

*   Unit tests - all table column resize tests are focused, so you can just run `yarn test -f table -wc`.
*   Manual tests:
    *   Primary focus should be put on http://localhost:8125/ckeditor5-html-support/tests/manual/ghs-all-features.html test. There is a table column resize feature included. Also there is already a resized table in the content. 
    *   Secondary focus - in http://localhost:8125/ckeditor5-table/tests/manual/tablecolumnresize.html I've adjusted it to test the case with enabling read-only that I described above - it:
        *   has a table with `<colgroup>` element but without `width` attribute,
        *   the editor turns into read-only mode after 2 seconds.

### CODE REVIEW

File: packages/ckeditor5-table/src/tablecolumnresize/tablecolumnresizeediting.js [hunk 1]
```diff
@@ -610,12 +609,15 @@ export default class TableColumnResizeEditing extends Plugin {
...
 						} else {
 							writer.removeStyle( 'width', viewFigure );
-							writer.removeClass(
-								'ck-table-resized',
-								[ ...viewFigure.getChildren() ].find( element => element.name === 'table' )
-							);
 						}
 					}
+
+					if ( !columnWidthsAttributeOld && !tableWidthAttributeOld ) {
```
Comment (mlewand) [inline packages/ckeditor5-table/src/tablecolumnresize/tablecolumnresizeediting.js:615]: We need explanation / reference for reason of this check.

**Review by mlewand — CHANGES_REQUESTED** [review]

**Comment by mlewand** [comment]
Somehow it didn't make into the review notes, sorry:

---

Let's add an integration test that will case from #12191.

So there should be call to `editor.setData( editor.getData() )` with GHS. I think that it makes sense to add this test in a separate test suite, because configuration will

**Review by mlewand — APPROVED** [review]
The solution looks good, however there are some DOM element leaks (see https://app.travis-ci.com/github/ckeditor/ckeditor5/jobs/579209722). Fixed in daac6ad.

It make sense to add column resize to the GHS manual test. Added in d11c8fc.

**Comment by Dumluregn** [comment]
>  however there are some DOM element leaks

Sorry, I was about to take care of it and forgot 👎🏻

**Comment by mlewand** [comment]
> > however there are some DOM element leaks
> 
> Sorry, I was about to take care of it and forgot 👎🏻 Thanks for taking care of it.

img

**Comment by mlewand** [comment]
Issue with

```
🔥 The following errors have been found:
 RESPONSE FAILURE   ➜  1 error
  
  ❌ Failed to load resource from upload.wikimedia.org (HTTP response status code: 404)
    
    …found on the following page:
    ➥  http://localhost:8125/ckeditor5/tests/manual/performance/editorinit.html
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Closing the manual test server...
```

Has been happening in epic (and probably master) branch already. Merging.

**Pull request merged**